src/material/slider/slider.ts

```diff
@@ -289,666 +345,1078 @@ export class MatSliderVisualThumb implements AfterViewInit, OnDestroy {
-    this._discrete = coerceBooleanProperty(v);
+
+  // TODO(wagnermaciel): describe the difference between inactive and active width and why we need it.
+  _updateWidthActive(): void {
+    this._paddingStyle = `0 ${this._slider._inputPadding}px`;
+    this._hostElement.style.width = `calc(100% - ${this._slider._inputPadding * 2}px)`;
   }
-  private _discrete: boolean = false;
 
-  /** Whether the slider displays tick marks along the slider track. */
-  @Input()
-  get showTickMarks(): boolean {
-    return this._showTickMarks;
+  // TODO(wagnermaciel): describe the difference between inactive and active width and why we need it.
+  _updateWidthInactive(): void {
+    this._paddingStyle = '0px';
+    this._hostElement.style.width = '100%';
   }
-  set showTickMarks(v: BooleanInput) {
-    this._showTickMarks = coerceBooleanProperty(v);
+
+  _updateThumbUIByValue(options?: {withAnimation: boolean}): void {
+    this.translateX = this._clamp(this._calcTranslateXByValue());
+    this._updateThumbUI(options);
   }
-  private _showTickMarks: boolean = false;
 
-  /** The minimum value that the slider can have. */
-  @Input()
-  get min(): number {
-    return this._min;
+  _updateThumbUIByPointerEvent(event: PointerEvent, options?: {withAnimation: boolean}): void {
+    this.translateX = this._clamp(this._calcTranslateXByPointerEvent(event));
+    this._updateThumbUI(options);
   }
-  set min(v: NumberInput) {
-    this._min = coerceNumberProperty(v, this._min);
-    this._reinitialize();
+
+  _updateThumbUI(options?: {withAnimation: boolean}) {
+    this._slider._transition =
+      options?.withAnimation && !this._slider._noopAnimations ? 'transform 80ms' : 'transform 0ms';
+    this._slider._onTranslateXChange(this);
+    this._slider._cdr.markForCheck();
   }
-  private _min: number = 0;
 
-  /** The maximum value that the slider can have. */
-  @Input()
-  get max(): number {
-    return this._max;
+  focus(): void {
+    this._hostElement.focus();
   }
-  set max(v: NumberInput) {
-    this._max = coerceNumberProperty(v, this._max);
-    this._reinitialize();
+
+  blur(): void {
+    this._hostElement.blur();
   }
-  private _max: number = 100;
+}
 
-  /** The values at which the thumb will snap. */
-  @Input()
-  get step(): number {
-    return this._step;
+@Directive({
+  selector: 'input[matSliderStartThumb], input[matSliderEndThumb]',
+  exportAs: 'matSliderRangeThumb',
+  providers: [],
+  host: {
+    '[style.pointer-events]': '_pointerEvents',
+    '[style.left]': '_left',
+    '[style.right]': '_right',
+    '[style.z-index]': '_zIndex',
+  },
+})
+export class MatSliderRangeThumb extends MatSliderThumb {
+  _pointerEvents: string = 'auto';
+  _left: string;
+  _right: string;
+  _marginLeft: string;
+  _marginRight: string;
+
+  getSibling(): MatSliderRangeThumb | undefined {
+    if (!this.__sibling) {
+      this.__sibling = this._slider._getInput(this._isEndThumb ? Thumb.START : Thumb.END) as
+        | MatSliderRangeThumb
```

Comment:
`as | MatSliderRangeThumb | undefined` - first `|` seems unnecessary, didn't even realize this was valid syntax

Reply:
This was done by `yarn format`

Reply:
Interesting, ok then its fine with me